### DISCUSSION TITLE
A few questions from a user fall in love with dmenu-extended

### DISCUSSION THREAD

**Discussion by Terkyz** [MarkHedleyJones/dmenu-extended#165] [General]
1. My default terminal is Alacritty and when I run a command, let's say 'ls' with a semicolon my terminal isn't opening, I tried configuring my dmenuExtended_preferences.txt file changing the terminal from xterm to alacritty, and it doesn't fix my problem.
2. How I can change the default image viewer (or in general, any default X viewer)? I know that I can use for example "nomacs: image.png" but when I directly tried to open a .png directly from dmenu-extended the default image viewer isn't nomacs
3. How often the cache is updated? Sometimes I download a file and when I quickly search it from dmenu-extended is not there yet.
4. I wish that some kind of 'Firefox bookmarks' plugin existed but I guess it doesn't.

Thank you SO much for creating dmenu-extended. :)

**Comment by markhedleyjones** [comment]
Hi @Terkyz,
Sorry for the delay in my response.
So I've just had a play around and here are some responses:
1. Normally, when a command exists the terminal will be closed. You're on the right track using a semi-colon to force the application to be run inside a terminal. Add another semi-colon (so `ls;;`) to have that terminal remain open once the program exists.
2. The way I change my default image viewer is to find an image file in Nautilus. Right click on it and choose "Open With", then select the application and select the "Always use for this filetype" option at the bottom of the dialog. There are other ways to do it depending on what environment you're using.
3. In general the cache is only updated when you ask for it to be updated. However, if you install the systemd service (instructions in the README.md) then it defaults to a period of 20 mins. Personally I don't use it as I don't like heavy things periodically running in the background.
4. Yeah that would really cool actually. If you have any questions about making your own plugins I'm happy to help where I can. It would be great to add that.
